Xamarin.Forms.Platform.Android/AppCompat/Platform.cs

```diff
@@ -220,10 +220,25 @@ internal void SetPage(Page newRoot)
 			var layout = false;
 			if (Page != null)
 			{
-				_renderer.RemoveAllViews();
+				var removeList = new List<global::Android.Views.View>();
+				for (int i = 0; i < _renderer.ChildCount; i++)
+				{
+					removeList.Add(_renderer.GetChildAt(i));
+				}
+
+				var renderersToDispose = _navModel.Roots.Select(Android.Platform.GetRenderer).ToList();
+
+				Device.StartTimer(TimeSpan.FromSeconds(1), () =>
```

Comment:
I wish the code had more comments for scenarios like this. Why 1 seconds. Why not 100 milliseconds.